src/transactions/conversion-check.guard.ts

```diff
@@ -1,29 +1,46 @@
-import {Injectable, CanActivate, ExecutionContext, UnauthorizedException, BadRequestException} from '@nestjs/common';
+import {
+	Injectable,
+	CanActivate,
+	ExecutionContext,
+	UnauthorizedException,
+	BadRequestException,
+	ForbiddenException
+} from '@nestjs/common';
+import {getRepository} from 'typeorm';
 import {SignedInBot} from 'types/bot';
+import {Currency} from 'src/currencies/currency.entity';
 import {Transaction} from './transaction.entity';
 
 @Injectable()
 export class ConversionCheckGuard implements CanActivate {
```

Comment:
Also, all this added logic should be in a validator file, not a guard. Relevant docs on [validation](https://docs.nestjs.com/techniques/validation) with `class-validator` and broader use [validation pipes](https://docs.nestjs.com/pipes) (more suited for this use case).